docs/development/CodingStyle.md

```diff
@@ -257,3 +257,34 @@ It should be based on sound mathematical, physical or computer science principle
 This is important for test code too. Tests shall be based on such principles and real-world properties so they don't just test the current implementation as it happens to be.
 
 Guidelines not tramlines: guidelines are not totally rigid - they can be broken when there is good reason.
+
+#Avoiding Accidental Mistakes
+
+* Always use braces when nesting if/else statements to avoid dangling else (the else branch belongs to the innermost if statement).
```

Comment:
IMO BF decided to use blocks for `if`  in all cases, so this is a bit redundant.

(Personally I prefer to skip braces in some cases:
```
if (test)
  return;
if (test)
  continue;
```
)

Reply:
I would prefer to use no braces in these two examples also. The rule is for nesting if/else statements, in which we might face dangling else problems. Thanks!

Reply:
@cpsw , @ledvinap ,the reason for always using braces is to do with viewing differences. If braces are always used and extra code is added, then the difference between the old and the new code is just the changed code. Braces are not always used then the difference also contains extra lines to do with the braces.

Although having a few extra lines with braces may not seem much, it makes the diffs bigger and adds up, especially with large PRs. Code reviews are easier if the review is not distracted by irrelevant changes.

Reply:
I am with @cpsw on this. In terms of maintainability, and the fact that code changes over time, it is easy to get from this case to a hard to detect bug.

```
if (test)
#if defined(SPECIAL_CASE)
  done = true;
#endif
  continue;
```